workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs

```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
 {
     public static class CommandLineUtility
     {
+        public static Dictionary<string, string> GetArguments()
+        {
+#if UNITY_ANDROID
```

Comment:
ifdefs like this in generic utility make me sad - in the long run we are going to have a lot of platform-dependent code where it shouldn't be.
The alternative is reworking our processing which would take some effort, so I'm fine with this for now, but we'll have to tackle that issue eventually.

Reply:
Sorry, misunderstood the goal of the code on the first read.
I'd still vote for restructuring this code and having dedicated class that takes care of mobile/android argument parsing and have here only call that class under ifdef

Reply:
I'm still against this as we don't have any mobile code in core and don't want to introduce it here

Reply:
Why not?

Reply:
I agree with @zeroZshadow here - this is a good place for "parse command line arguments for any platform", and it's pretty much the standard way of doing these kinds of things.

Reply:
As the core should only provide the bare minimum to ensure that you can communicate with SpatialOS. If you want to use mobile, you need the mobile feature module anyway, so let's try to keep all mobile related code in one place instead of spreading it all over the place.

(there is also a case for getting rid of this utility and inlining it in the defaultworkerconnector, but that's a different discussion)

Reply:
I'd argue for a middle ground here. On one hand, I don't think this belongs in mobile module, but rather in utility module - if we throw everything used in mobile in one module, we'll end up with a mobile module being a bag of logically unrelated things. On the other hand, this code looks a bit alien here - I would separate it in another method (like ExtractAndroidArguments) and call it here under ifdef. Potentially in the future we might want to create a separate class or dedicated mobile utility module

Reply:
while this sounds better, I don't think we should create a new module for one script. The mobile module is already very sparse and contains mainly utility functionality to make it easier, so I don't see a reason why we shouldn't have this method in there as well

Reply:
splitting functionality into different focused module is good, but not if the module is still really small and there is no added benefit in doing it so. We can think about refactoring things out when the module becomes too big